src/node/utils/caching_middleware.js

```diff
@@ -196,9 +196,12 @@ CachingMiddleware.prototype = new function () {
         var statusCode = responseCache[cacheKey].statusCode;
 
         var pathStr = CACHE_DIR + 'minified_' + cacheKey;
-        if (supportsGzip && (headers['content-type'] || '').match(/^text\//)) {
+        if (supportsGzip && (headers['content-type'] || '').match(/^application\/javascript/)) {
           pathStr = pathStr + '.gz';
           headers['content-encoding'] = 'gzip';
+        } else {
+          // ensure responseCache is updated
+          delete headers['content-encoding'];
```

Comment:
Rather than clear it here, wouldn't it be better to not store the header in the cache to begin with? I think that would make the code easier to understand.

Reply:
I tried it in https://github.com/ether/etherpad-lite/pull/4491/commits/142cf0b6301e7bbfd275b4fb00f9d351afe02350